### ISSUE TITLE
AnkiDroid Directory not able to be accessed 

### ISSUE THREAD

**Issue by Raphaelpp** [ankidroid/Anki-Android#5314]
###### Reproduction Steps

On AnkiDroid (honor 9 lite phone) 

1. Open "préférence" I have a French version (so settings I guess) 
2. Sentence "le répertoire dankidroid est inaccessible" (AnkiDroid Directory not able to be accessed) . 
3. If I go back, the app closes it self. 
![Screenshot_20190527-095549](https://user-images.githubusercontent.com/51075268/58404537-a3393b80-8065-11e9-87fe-ad8dfc7426b3.jpg)

Thanks for helping me, I suppose I have to re-write the file access which is actullay "/sdcard/AnkiDroid", what should I do? 

I have an account and ofte make  synchronisations, but I don't want to take the risk ti unistall (I have 600 cards). 

Best regards. 

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [ x ] character to confirm the points below:*

[  ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[  ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[  ] I have searched for similar existing issues here and on the user forum

**Comment by mikehardy** [comment]
I'm not sure we have this issue formally tracked anywhere so this might be the first entry in github, but we have been seeing this for non-english phones going into advanced preferences.

We are not sure the cause of it, so I appreciate the details here, maybe this will help pin it down.

In the meantime the workaround is to uninstall and reinstall the app. It won't affect your collection (it is in /sdcard/AnkiDroid and will remain there during the uninstall / reinstall) but for some reason it makes it so that this setting is no longer corrupted

**Comment by mikehardy** [comment]
For 2.9 I think this is now our most critical outstanding bug.
This happens to lots of people. It appears to only happen to non-English languages.

Reproduction should be:
- make sure your language is not english and has ankidroid translations
- go into advanced preferences (I think this is where the damage is done - similar to #5346 - just opening it changes prefs state)
- restart the app and it fails

It's a terrible user experience but I haven't had time to look into it yet

**Comment by mikehardy** [comment]
I should mention it results in around 2-3 requests for help on the play store, per week, for the 2.8 version (not sure it still applies in 2.9 so maybe test 2.8 first?) which is the most frequent error reported there now.

If anyone wants to check v2.8.4 I have a version that compiles locally here: https://github.com/mikehardy/Anki-Android/tree/v2.8.4-buildable (that's necessary because since v2.8.4 was released long ago, there have been such large changes in AndroidStudio that you can't build it out of the box anymore, you need the changes I made here https://github.com/mikehardy/Anki-Android/commit/d10d3dce25172fe968533b558ccde05c348085cc

**Comment by quentin-moquay** [comment]
I'm a french user. 

I tried to update a lot of settings in the same screen but I was not able to reproduce it.

Some infos:
* I'm using the AnkiDroid from Play Store, localization : French
* My phone is an Xiaomi Redmi Note 5, Android 9 PKQ1.180904.001
* I've got an external SD Card but Ankidroid folder is on internal storage
* There is more than 5Gb free storage right now
* I have Aedict3 installed
* More than 40 000 cards on my collection

I hope we could find something discriminant to know how to reproduce it. Maybe I'm forgetting to give more information, don't hesitate to ask me more.
